subsys/testsuite/ztest/src/ztest_new.c

```diff
@@ -464,6 +476,42 @@ struct ztest_unit_test *ztest_get_next_test(const char *suite, struct ztest_unit
 	return NULL;
 }
 
+#if (IS_ENABLED(CONFIG_ZTEST_DO_THE_SHUFFLE))
+static void ztest_seed_the_shuffle()
+{
+	uint32_t z_seed = SHUFFLE_SEED;
+
+	if (!z_seed){
+#if (IS_ENABLED(CONFIG_POSIX_CLOCK))
+		time_t t = 0;
+		z_seed = (uint32_t)time(&t);
+#else
+		z_seed = k_cycle_get_32();
+#endif
+	}
+
+
+	PRINT("Let's shuffle! seeding: %u\n", z_seed);
+	srand(z_seed);
+}
+
+static void z_ztest_shuffle(void *array, size_t num_items, size_t elem_size)
+{
+	char tmp[elem_size];
```

Comment:
How does this work? I don't think you can declare this as a dynamic sized array. Can we just pass `void *tmp` and `size_t elem_size` into the function?

Reply:
This is considered a Variable Length Array supported in C99 standard. I can tweak this to malloc temp space.  I don't think the caller of the function should be allocating a buffer to handle swapping.